### ISSUE TITLE
[Feature] configurable screen curvature [sf#236]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/openMSX#260]
**Reported by eadmaster on 2012-01-17 19:33 UTC**
The display_deform 3d effect looks too much "inclined", i think it should not be like that.

**Comment by openMSX-import** [comment]
**Commented by eadmaster on 2012-01-18 20:16 UTC**
i mean, the MSX is not an arcade machine, the display should not be inclined at all!

**Comment by openMSX-import** [comment]
**Updated by eadmaster on 2012-01-18 20:16 UTC**
- **labels**:  --> General/misc

**Comment by oh-ren** [comment]
I agree with that, it should give the feeling you're in front of a CRT, but IMO this feature could/should be replaced by the ability to load/apply GLSL shaders (see e.g. MAME/ RetroArch)?

Related: #975

**Comment by oh-ren** [comment]
Perhaps an option could be added to `display_deform` that gives a more realistic MSX monitor effect.
I guess this is a cost-effective way to extend openMSX's current CRT simulation options?

e.g. `set display_deform arcade` (the current effect) & `set display_deform monitor` (the new option)

(ps. as this is an @openMSX-import issue, I'm not sure if anyone is notified when such an issue is commented on, @MBilderbeek? (sorry, no intrusiveness intended ;))

**Comment by oh-ren** [comment]
So, openMSX devs, what say you?
Small adjustment, lots of pleasure (perhaps).
I wonder who actually uses the arcade 3d look anyway, besides a little tryout / playing around, whilst my suggestion _might_ lead to the feature actually used & enjoyed? Cheers! :+1: :beers:

**Comment by MBilderbeek** [comment]
It's not clear to me what exactly you are proposing...

**Comment by oh-ren** [comment]
You mean it's **not** clear?

**Comment by MBilderbeek** [comment]
Indeed, fixed comment :)

**Comment by oh-ren** [comment]
It's all in the post:
`set display_deform monitor` should result in an effect/geometry that looks more like [this](https://www.youtube.com/watch?v=gPwZO8EL62A) (perhaps a little bit of [inclination](https://www.youtube.com/watch?v=D5nhpJfaF_U) _would_ be cool) instead of the big, arcade cabinet like effect/inclination the current deform gives, so it actually resembles more of sitting in front of a (MSX) CRT monitor.
The current effect could still be toggled via  the new `set display_deform arcade`.

For the time being I figured it's perhaps something nice to implement with little effort (just adjust some vars?) And if you guys feel like it, one day you might implement something else (better / more powerful & configurable) to handle this stuff.. :)

The mode could even implement/show the monitor bezel/frame.. As a nice cool extra.. :sunglasses: ([arcade example](https://www.youtube.com/watch?v=ZvSrrdV5VeU)) (that's a bit 'heavy'/much for our purposes though I reckon.. ;))

**Comment by MBilderbeek** [comment]
OK, well, I don't know much about shader programming, so I'll leave this up to Wouter.

**Comment by oh-ren** [comment]
Any interest in picking this up?
